src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class SettingsStorageHelper
+{
+public:
+    SettingsStorageHelper(const QString &keyName)
+        : m_keyName(keyName)
+    {
+    }
+
+    void set(const T &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    T get(const QVariant &defaultValue = QVariant())
```

Comment:
>can you call QVariant::value<T>() for an empty QVariant? 

From Qt doc: http://doc.qt.io/qt-4.8/qvariant.html#value
>If the value cannot be converted, default-constructed value will be returned.

That's the worst case, so I think it will work fine.

-
>Would it be better to rephrase it in the following way:

I think no, its probably better to try matching the `QVariant` ctor signature earlier rather than later, in case of an incompatible type results in incomprehensible compiler message.